controllers/helpers_test.go

```diff
@@ -162,23 +162,19 @@ func mockedCreateLBCalls(t *testing.T, m *mock_elbiface.MockELBAPIMockRecorder)
 		LoadBalancerName: aws.String(""),
 	})).MaxTimes(1)
 
-	m.AddTags(gomock.Eq(&elb.AddTagsInput{
-		LoadBalancerNames: aws.StringSlice([]string{""}),
-		Tags:              expectedTags,
-	})).Do(func(actual *elb.AddTagsInput) (*elb.AddTagsOutput, error) {
-		sortTagsByKey := func(tags []*elb.Tag) {
-			sort.Slice(tags, func(i, j int) bool {
-				return *(tags[i].Key) < *(tags[j].Key)
-			})
-		}
+	m.AddTags(gomock.AssignableToTypeOf(&elb.AddTagsInput{})).Return(&elb.AddTagsOutput{}, nil).Do(
+		func(actual *elb.AddTagsInput) {
+			sortTagsByKey := func(tags []*elb.Tag) {
+				sort.Slice(tags, func(i, j int) bool {
+					return *(tags[i].Key) < *(tags[j].Key)
+				})
+			}
 
-		sortTagsByKey(actual.Tags)
-		sortTagsByKey(expectedTags)
```

Comment:
Why don't we sort the expected keys?

What is the fix in this PR?

Reply:
I have sorted it manually in the declaration itself, only change is we are expecting AddTagsInput only with type check now and not with the actual input fields.

Reply:
This has been followed in other files as well, so followed the same, and those tests doesn't flake.

Reply:
The sorttags func works fine, i think the main catch here is the expected mock input for AddTagsInput. It used to first compare the expectedTags and then do sorting, that's why it used to fail sometimes, now we dont rely on the input parameters itself and just check AssignableOfType AddTagsInput.

Reply:
So even if we leave th sortTags(expectedTags) as is, it will work.